alameda.js

```diff
@@ -111,10 +111,10 @@ var requirejs, require, define;
         shim: {},
         config: {}
       },
-      mapCache = {},
+      mapCache = Object.create(null),
       requireDeferreds = [],
-      deferreds = {},
-      calledDefine = {},
+      deferreds = Object.create(null),
+      calledDefine = Object.create(null),
       calledPlugin = {},
```

Comment:
IE 10 and 11 support Object.create (which is important for [alameda-prim](https://github.com/requirejs/alameda-prim), so let's try to land this as-is. I may do some extra cleanup and maybe even try to remove the hasProp checks completely, but let's get this baseline in first.

There is a conflict with alameda.js at the moment in this PR branch. Is that something you would like to resolve/have time for?